addon/helpers/filter-by.js

```diff
@@ -0,0 +1,57 @@
+import Ember from 'ember';
+
+const {
+  Helper,
+  computed: { filter, filterBy },
+  defineProperty,
+  get,
+  isArray,
+  isPresent,
+  observer,
+  set
+} = Ember;
+
+export default Helper.extend({
+  compute([array, byPath, value]) {
+    set(this, 'array', array);
+    set(this, 'byPath', byPath);
+    set(this, 'value', value);
+
+    return get(this, 'content');
+  },
+
+  array:   null,
+  byPath:  null,
+  value:   null,
+  content: null,
+
+  byPathDidChange: observer('byPath', 'value', function() {
+    let byPath = get(this, 'byPath');
+    let value = get(this, 'value');
+
+    if (byPath) {
+      let filterFn;
+
+      if (value) {
+        if (typeof value === 'function') {
+          filterFn = (item) => value(get(item, byPath));
+        } else {
+          filterFn = (item) => get(item, byPath) === value;
+        }
+      } else {
+        filterFn = (item) => isPresent(get(item, byPath));
+      }
+
+      let cp = filter(`array.@each.${byPath}`, filterFn);
+
+      defineProperty(this, 'content', cp);
+    } else {
+      defineProperty(this, 'content', []);
```

Comment:
How about we move this to the top to return early, so we don't need an `if/else` and just an `if`:

```js
if (isEmpty(byPath)) {
  defineProperty(this, 'content', []);
  return;
}

let filterFn;

if (value) {
  ...
}
```